INSTALL: document Python 3 as a build tool

cjmayo opened the pull request:
Needed for scripts/process_function_template.py run by
src/core/CMakeLists.txt.

Enforced since commit:
f16fe73476 ("Python is a hard build time dependency", 2015-09-20)

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

Gustry commented:
INSTALL is an output file and shouldn't be edited. You should edit in the doc folder

cjmayo committed:
INSTALL: document Python 3 as a build tool

Needed for scripts/process_function_template.py run by
src/core/CMakeLists.txt.

Enforced since commit:
f16fe73476 ("Python is a hard build time dependency", 2015-09-20)

cjmayo commented:
> INSTALL is an output file and shouldn't be edited. You should edit in the doc folder

Oops, apologies, forgot that. Edited doc/overview.t2t and regenerated INSTALL and doc/INSTALL.html.
(tweaked commit title too).

m-kuhn commented:
Thank you
